version.go

```diff
@@ -137,9 +137,30 @@ func (v *Version) IsK0s() bool {
 
 // K0sVersion returns the k0s version (eg 4 from v1.2.3-k0s.4)
 func (v *Version) K0sVersion() int {
```

Comment:
A safer to use signature would be `func (v *Version) K0sVersion() (uint, bool)`. I know that unsigned integers are always treated a bit shabbily in the Go world, but using negative values as an indicator for errors is something I can just about do for historical reasons with array index functions. I imagine the non-negative check will be forgotten at call sites, yielding strange results, whereas everybody will check what that second bool return value is used for. :sweat_smile: